plugins/modules/identity/keycloak/keycloak_user_info.py

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2019, Adam Goossens <adam.goossens@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_user_info
+
+short_description: Allows obtaining Keycloak user information via Keycloak API
+
+version_added: 5.4.0
+
+description:
+    - This module allows you to get Keycloak user information via the Keycloak REST API.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+
+    - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+      be returned that way by this module. You may pass single values for attributes when calling the module,
+      and this will be translated into a list suitable for the API.
+
+options:
+    auth_keycloak_url:
+        description:
+            - URL to the Keycloak instance.
+        type: str
+        required: true
+        aliases:
+          - url
+    validate_certs:
+        description:
+            - Verify TLS certificates (do not disable this in production).
+        type: bool
+        default: yes
+
+    realm:
+        type: str
+        description:
+            - They Keycloak realm ID.
+        default: 'master'
+    id:
+        type: str
+        description:
+            - They Keycloak User ID.
+    name:
+        type: str
+        description:
+            - They Keycloak User Name.
+
+author:
+    - Dishant Pandya (@drpdishant)
+    - Mahek Katariya (@MahekKatariya)
+
+extends_documentation_fragment:
+- community.general.keycloak
+'''
+
+EXAMPLES = '''
+- name: Get all users in a realm
+  community.general.keycloak_user_info:
+    realm: MyCustomRealm
+    auth_keycloak_url: https://auth.example.com/auth
+  delegate_to: localhost
+
+- name: Get a Keycloak User Info by User Id
+  community.general.keycloak_user_info:
+    realm: MyCustomRealm
+    userid: "677a27df-02e8-48e9-8f90-fa95caf64bd7" #UUID of the User
+    auth_keycloak_url: https://auth.example.com/auth
+  delegate_to: localhost
+
+- name: Get a Keycloak User Info by User Name
+  community.general.keycloak_user_info:
+    realm: MyCustomRealm
+    username: "user@example.com"
+    auth_keycloak_url: https://auth.example.com/auth
+  delegate_to: localhost
+'''
+
+RETURN = '''
+msg:
+    description: Message as to what action was taken.
+    returned: always
+    type: str
+
+end_state:
```

Comment:
I don't think `end_state` is an appropriate return value name for an `_info` module.